src/components/company/CompanySocialImpactComparison.vue

```diff
@@ -105,40 +97,41 @@ export default {
           CompaniesRepository.triggerServerComputation(this.companyData.denomination.value, year);
         }
       }
-      console.log(" - - CompanySocialImpactComparison statistics :", result);
-      return result;
     },
-    apeCodes: function () {
-      var companyApe = this.companyData.ape.code;
-      while (companyApe.length < 4) {
-        companyApe = "0" + companyApe;
+    initializeStatistics: async function () {
+      // Get scores description
+      let result = await OntologyRepository.getScores();
+      for (var statType in result) {
+        result[statType].values = {};
       }
-      return [companyApe, companyApe.substring(0, 3), companyApe.substring(0, 2)];
+      this.statistics = result;
+      this.fillStatistics();
```

Comment:
j'imaginais plutot qq chose comme ca:
```js
this.statistics = this.buildStatistics();
```
ca permet de construire l'état avant de le calculer. c'est plus propre et surtout ca ne déclenche pas de modification de l'interface pour chaque changement.
et this.buildStatistics(); pourait devenir indépendent du composant car sans référence à this.statistics